File: task/task-list-iterative-pagination.py
from twelvelabs import TwelveLabs
import os

client = TwelveLabs(api_key=os.environ.get("API_KEY"))

def print_page(page):
    for task in page:
      print(f"Task ID: {task.id}")
      print(f"Index ID: {task.index_id}")
      print(f"Video ID: {task.video_id}")
      print(f"Estimated time: {task.estimated_time}")
      print(f"Status: {task.status}")
      print("Metadata:")
      for key, value in task.metadata.items():
          print(f"  {key}: {value}")

      if task.hls:
          print("HLS:")
          print(f"  Video URL: {task.hls.video_url}")
          print("  Thumbnail URLs:")
          for url in task.hls.thumbnail_urls or []:
              print(f"    {url}")
          print(f"  Status: {task.hls.status}")
          print(f"  Updated at: {task.hls.updated_at}")

      if task.process:
          print("Process:")
          print(f"  Percentage: {task.process.percentage}%")
          print(f"  Remaining Seconds: {task.process.remain_seconds}")

      print(f"Created at: {task.created_at}")
      print(f"Updated at: {task.updated_at}")

# Fetch the initial page of results
task_paginator = client.task.list_pagination(
  id="<YOUR_TASK_ID>",
  index_id="<YOUR_INDEX_ID>",
  filename="<YOUR_FILENAME>",
  duration=20,
  width=1920,
  height=1080,
  sort_by = "updated_at",
  sort_option="asc",
  created_at="2024-09-17T07:53:46.365Z",
  updated_at="2024-09-17T07:53:46.365Z",
  estimated_time="2024-09-17T07:55:22.125Z",
  page=2,
  page_limit=5,
)

# Print the first page of results
print_page(task_paginator.data)

# Iterate through subsequent pages
while True:
    try:
        next_task_page = next(task_paginator)
        print_page(next_task_page)
    except StopIteration:
        break
    

File: task/task-list-iterative-pagination.js
import { TwelveLabs } from 'twelvelabs-js';

function printPage(page) {
  page.forEach(task => {
    console.log(`Task ID=${task.id}`);
    console.log(`Index ID: ${task.indexId}`);
    console.log(`Video ID: ${task.videoId}`);
    console.log(`Estimated time: ${task.estimatedTime}`);
    console.log(`Status: ${task.status}`);
    
    console.log("Metadata:");
    for (const [key, value] of Object.entries(task.metadata)) {
      console.log(`  ${key}: ${value}`);
    }
  
    if (task.hls) {
      console.log("HLS:");
      console.log(`  Video URL: ${task.hls.videoUrl}`);
      console.log("  Thumbnail URLs:");
      for (const url of task.hls.thumbnailUrls || []) {
        console.log(`    ${url}`);
      }
      console.log(`  Status: ${task.hls.status}`);
      console.log(`  Updated at: ${task.hls.updatedAt}`);
    }
  
    if (task.process) {
      console.log("Process:");
      console.log(`  Percentage: ${task.process.percentage}%`);
      console.log(`  Remaining Seconds: ${task.process.remainSeconds}`);
    }
  
    console.log(`Created at: ${task.createdAt}`);
    console.log(`Updated at: ${task.updatedAt}`);
  });
}


(async () => {
  const client = new TwelveLabs({ apiKey: process.env.API_KEY}); 

  const listParams = {
    id: '<YOUR_TASK_ID>',
    index_id: '<YOUR_INDEX_ID>',
    filename: '<YOUR_FILENAME>',
    duration: 20,
    width:1920,
    height: 1080,
    sortBy: 'updated_at',
    sortOption: 'asc',
    // createdAt: '2024-09-17T07:53:46.365Z',
    // updatedAt: '2024-09-17T07:53:46.365Z',
    estimatedTime: '2024-09-17T07:55:22.125Z',
    page: 2,
    pageLimit: 5,
  };

  // Fetch the initial page of results
  const taskPaginator = await client.task.listPagination(listParams);

  // Print the first page of results
  printPage(taskPaginator.data);

  // Iterate through subsequent pages
  while (true) {
    const nextTaskPage = await taskPaginator.next();
    if (!nextTaskPage) {
      console.log('No more pages of index results available');
      break;
    }
    printPage(nextTaskPage);
  }
})();
